DigiMeSDK/Core/Classes/Security/DMEJWTManager.swift

```diff
@@ -0,0 +1,444 @@
+        let isVerified = JWT<PayloadRefreshOAuthJWT>.verify(signedToken, using: verifier)
+
+        return isVerified ? signedToken : nil
+    }
+    
+    // MARK: - Utility functions
+    private func convertKeyString(_ keyString: String) -> Data? {
+        guard let base64String = try? CryptorRSA.base64String(for: keyString) else {
+            print("Couldn't read a key string")
+            return nil
+        }
+
+        return convertKeyBase64ToData(base64String)
+    }
+    
+    private func convertKeyBase64ToData(_ base64KeyString: String) -> Data? {
+        guard let publicKeyData = Data(base64Encoded: base64KeyString, options: [.ignoreUnknownCharacters]) else {
+            print("Couldn't decode base64 key")
+            return nil
+        }
+
+        return publicKeyData
+    }
+}
+
+// MARK: - Utility extensions
+
+extension DMEJWTManager {
+    static let authorityPublicKey: String = "authorityPublicKey"
+    static let codeVerifier: String = "codeVerifier"
+    
+    struct AuthorityPublicKey: Codable {
+        var publicKeyBase64: String
+        var creationDate: Date
+    }
+    
+    // we need to save Authority public key in the case if app will be restarted
+    func saveAuthorityPublicKey(_ publicKeyBase64: String, creationDate: Date) {
+        let data = AuthorityPublicKey(publicKeyBase64: publicKeyBase64, creationDate: creationDate)
+        let encoder = JSONEncoder()
+        if let encoded = try? encoder.encode(data) {
+            let defaults = UserDefaults.standard
+            defaults.set(encoded, forKey: DMEJWTManager.authorityPublicKey)
+        }
+    }
+    
+    // we only reuse stored authority public key if it was stored less than 15 mins ago
+    func retrieveAuthorityPublicKeyIfValid() -> String? {
+        if let data = UserDefaults.standard.object(forKey: DMEJWTManager.authorityPublicKey) as? Data {
+            if
+                let decoded = try? JSONDecoder().decode(AuthorityPublicKey.self, from: data),
+                let diff = Calendar.current.dateComponents([.minute], from: decoded.creationDate, to: Date()).minute,
+                diff < 15 {
+                    return decoded.publicKeyBase64
+            }
+        }
+        
+        return nil
+    }
+    
+    // save code verifier for OAuth session
+    func saveCodeVerifier(_ codeVerifier: String) {
+        let defaults = UserDefaults.standard
+        defaults.set(codeVerifier, forKey: DMEJWTManager.codeVerifier)
+    }
+    
+    func retrieveCodeVerifier() -> String? {
+        return UserDefaults.standard.object(forKey: DMEJWTManager.codeVerifier) as? String
+    }
+}
+
+extension Data {
+    init?(base64URLEncoded string: String) {
+        self.init(base64Encoded: string.convertToBase64())
+    }
+
+    func base64URLEncodedString() -> String {
+        return self.base64EncodedString().convertToBase64URL()
+    }
+}
+
+extension String {
+    func convertToBase64() -> String {
```

Comment:
We can drop the `convertTo` from the function name as when it is used in context the conversion part is already implied, eg `let val = myString.base64()` already implies that returned `val` will be in `base64` format.